bindings/python/tests/test_services.py

```diff
@@ -23,191 +23,211 @@
-        size = len(content)
-        await self.async_operator.write(filename, content)
-        metadata = await self.async_operator.stat(filename)
-        assert metadata is not None
-        assert metadata.mode.is_file()
-        assert metadata.content_length == size
-
-        await self.async_operator.delete(filename)
-
-    def test_sync_write_with_non_ascii_name(self):
-        size = randint(1, 1024)
-        filename = f"❌😱中文_{str(uuid4())}.test"
-        content = os.urandom(size)
-        size = len(content)
-        self.operator.write(filename, content)
-        metadata = self.operator.stat(filename)
-        assert metadata is not None
-        assert metadata.mode.is_file()
-        assert metadata.content_length == size
-
-        self.operator.delete(filename)
-
-    @pytest.mark.asyncio
-    async def test_async_write_with_non_ascii_name(self):
-        size = randint(1, 1024)
-        filename = f"❌😱中文_{str(uuid4())}.test"
-        content = os.urandom(size)
-        size = len(content)
-        await self.async_operator.write(filename, content)
-        metadata = await self.async_operator.stat(filename)
-        assert metadata is not None
-        assert metadata.mode.is_file()
-        assert metadata.content_length == size
-
-        await self.async_operator.delete(filename)
-
-    def test_sync_create_dir(self):
-        path = f"test_dir_{str(uuid4())}/"
-        self.operator.create_dir(path)
-        metadata = self.operator.stat(path)
-        assert metadata is not None
-        assert metadata.mode.is_dir()
-
-        self.operator.delete(path)
-
-    @pytest.mark.asyncio
-    async def test_async_create_dir(self):
-        path = f"test_dir_{str(uuid4())}/"
-        await self.async_operator.create_dir(path)
-        metadata = await self.async_operator.stat(path)
-        assert metadata is not None
-        assert metadata.mode.is_dir()
-
-        await self.async_operator.delete(path)
-
-    def test_sync_delete(self):
-        size = randint(1, 1024)
-        filename = f"test_file_{str(uuid4())}.txt"
-        content = os.urandom(size)
-        size = len(content)
-        self.operator.write(filename, content)
-        self.operator.delete(filename)
-        with pytest.raises(FileNotFoundError):
-            self.operator.stat(filename)
-
-    @pytest.mark.asyncio
-    async def test_async_delete(self):
-        size = randint(1, 1024)
-        filename = f"test_file_{str(uuid4())}.txt"
-        content = os.urandom(size)
-        size = len(content)
-        await self.async_operator.write(filename, content)
-        await self.async_operator.delete(filename)
-        with pytest.raises(FileNotFoundError):
-            await self.operator.stat(filename)
-
-
-class TestS3(AbstractTestSuite):
-    service_name = "s3"
-
-
-class TestFS(AbstractTestSuite):
-    service_name = "fs"
-
-
-class TestMemory(AbstractTestSuite):
-    service_name = "memory"
+
+@pytest.fixture()
```

Comment:
Thanks for the PR, but the class way is preferred since we want to support `pytest -vk memory,s3,fs` which means we could construct more operators.

Reply:
OKKKK，the function based can do the same thing, wait a liite bit

Reply:
> the function based can do the same thing

I don't understand why we need to completely change our approach just to accomplish the same thing. Are there any other benefits of using function-based tests?

Reply:
![CleanShot 2023-10-13 at 11 53 24@2x](https://github.com/apache/incubator-opendal/assets/7054676/e060250a-9137-4045-b663-f54485331662)

Reply:
But having a class make it more clear about the tests belongings:

![image](https://github.com/apache/incubator-opendal/assets/5351546/b363c524-1e9a-4e37-a4e7-705888707c68)